docs/orchestration/agents/kubernetes.md

```diff
@@ -1,202 +1,234 @@
+You can use the `--service-account-name` option to configure a [service
+account](https://kubernetes.io/docs/reference/access-authn-authz/service-accounts-admin/)
+to use for all Jobs deployed by the agent.
 
-```
-kubectl apply --namespace=AGENT_NAMESPACE -f -
+```bash
+prefect agent kubernetes start --service-account-name my-account
 ```
 
-:::
+Flows can override this agent default by passing the `service_account_name` option to
+their respective
+[KubernetesRun](/orchestration/flow_config/run_configs.md#kubernetesrun)
+`run_config`.
 
-Now you should be able to see the agent deployment created and running on your cluster:
+### Image Pull Secrets
 
-```
-$ kubectl get deploy
-NAME            DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
-prefect-agent   1         1         1            0           2s
+Likewise, the `--image-pull-secrets` option can be used to configure [image
+pull
+secrets](https://kubernetes.io/docs/concepts/containers/images/#specifying-imagepullsecrets-on-a-pod)
+for all Jobs deployed by the agent. Multiple secrets can be provided as a
+comma-separated list. Note that the secrets must already exist within the same
+`namespace` as the agent.
 
-$ kubectl get pods
-NAME                             READY   STATUS    RESTARTS   AGE
-prefect-agent-845798bb59-s7wxg   1/1     Running   0          5s
+```bash
+prefect agent kubernetes start --image-pull-secrets secret-1,secret-2
 ```
 
-You are now ready to run some flows!
+Flows can override this agent default by passing the `image_pull_secrets` option to
+their respective
+[KubernetesRun](/orchestration/flow_config/run_configs.md#kubernetesrun)
+`run_config`.
 
-#### Permissions
+### Custom Job Template
 
-If you are using Amazon EKS for your kubernetes deployment and you need S3 access, note that S3 is not accessible by default to Amazon EKS. To enable S3 access by your kubernetes cluster on EKS, add the necessary permissions (AmazonS3FullAccess or AmazonS3ReadOnlyAccess) directly to the NodeInstanceRole used by aws-auth-cm.yaml after launching worker nodes and before applying aws-auth-cm.yaml with kubectl.
+For deeper customization of the Kubernetes Job created by the agent, you may
+want to make use of a custom job template. This template can either be
+configured per-flow (on the
+[KubernetesRun](/orchestration/flow_config/run_configs.md#kubernetesrun)
+`run_config`), or on the Agent as a default for flows that don't provide their
+own template.
 
-#### Labels
+For reference, the default template packaged with Prefect can be found
+[here](https://github.com/PrefectHQ/prefect/blob/master/src/prefect/agent/kubernetes/job_template.yaml).
 
-To specify a set of labels for a Kubernetes Agent during install you may specify various `--label` arguments.
+To provide your own job template, you can use the `--job-template` flag. This
+takes a path to a job template YAML file. The path can be local to the agent,
+or stored in cloud storage on either GCS or S3.
```

Comment:
Made a small note. We'll need a better docs section for this in general (most user will probably want to store the aws/gcp credentials in the standard ways for those tools (e.g. `~/.aws/config` files) rather than secrets). I made a small comment saying "credentials are needed", but we'll want to amend this soon with a reference to a nicer docs section on this. Punting on it for now.

Reply:
https://github.com/PrefectHQ/prefect/issues/3788